tests/models/test_decoder.py

```diff
@@ -267,6 +269,86 @@ def test_attn_mask_construction(self, decoder, sample_decoder_data):
             assert not attn_mask[0, 1, 3]
             assert not attn_mask[1, 4, 5]
 
+    def test_flex_local_ca_mask_equivalence(self):
```

Comment:
Could we move these to be standalone tests under `tests/flex/`? Better to avoid couplng to the decoder class here if we just want to test the mask correctness (if we wanted to test the decoder integration then I think a test setup comparing two decoders, one with torch attention and one with flex and running allclose on the outputs would make more sense)